src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs

```diff
@@ -204,6 +207,78 @@ private async Task<DataServiceFeatureMatrix> GetFeatureMatrixInternal( FetchBeha
 			return _FeatureMatrix;
 		}
 
+		/// <summary>
+		/// Limit and sort received measurements if the initial request was split into multiple requests due to uri length.
+		/// </summary>
+		/// <param name="source">Fetched measurements</param>
+		/// <param name="filter">Set filter to check for LimitResult and OrderBy</param>
+		/// <param name="resultSets">Number of requests after splitting the initial request</param>
+		/// <returns>Array of measurements in correct order and size.</returns>
+		private static SimpleMeasurementDto[] LimitAndSortResult( IEnumerable<SimpleMeasurementDto> source, AbstractMeasurementFilterAttributesDto filter, int resultSets )
+		{
+			if( resultSets <= 1 ) return source.ToArray();
+
+			if( filter.OrderBy == null || !filter.OrderBy.Any() )
+				return filter.LimitResult < 0 ? source.ToArray() : source.Take( filter.LimitResult ).ToArray();
+
+			IOrderedEnumerable<SimpleMeasurementDto> orderedResult = null;
+			var firstFilter = filter.OrderBy.First();
+			var sourceArray = source.ToArray();
+
+			//Fast pass for default sorting by time
+			if( filter.OrderBy.Length == 1 && firstFilter.Attribute == WellKnownKeys.Measurement.Time )
+			{
+				orderedResult = firstFilter.Direction switch
+				{
+					OrderDirectionDto.Asc => sourceArray.OrderBy( m => m.Time ),
+					OrderDirectionDto.Desc => sourceArray.OrderByDescending( m => m.Time ),
+					_ => throw new ArgumentOutOfRangeException( nameof( firstFilter.Direction ) )
+				};
+			}
+
+			//Evaluate each OrderBy
+			else
+			{
+				foreach( var orderDto in filter.OrderBy )
+				{
+					orderedResult = OrderByAttribute( orderDto, orderedResult, sourceArray );
+				}
+			}
+
+			if( orderedResult == null )
+				return Array.Empty<SimpleMeasurementDto>();
+
+			return filter.LimitResult < 0 ? orderedResult.ToArray() : orderedResult.Take( filter.LimitResult ).ToArray();
+		}
+
+		/// <summary>
+		/// Sort the measurements according to the order criteria. Chain it to previousResult or start an new chain using source if previousResult is null.
+		/// </summary>
+		private static IOrderedEnumerable<SimpleMeasurementDto> OrderByAttribute( OrderDto order, IOrderedEnumerable<SimpleMeasurementDto> previousResult, IEnumerable<SimpleMeasurementDto> source )
+		{
+			IOrderedEnumerable<SimpleMeasurementDto> result;
+			if( previousResult == null )
+			{
+				result = order.Direction switch
+				{
+					OrderDirectionDto.Asc => source.OrderBy( m => m.Attributes.FirstOrDefault( a => a.Key == order.Attribute )?.Value ),

```

Comment:
Ich würde `m.Attributes.FirstOrDefault( a => a.Key == order.Attribute )?.Value` in eine Methode auslagern, die auch die Weiche für `Time` enthält.

Reply:
Erledigt